src/main/java/com/compilercharisma/chameleonbusinessstudio/repository/AppointmentRepositoryv2.java

```diff
@@ -25,32 +25,49 @@
 public class AppointmentRepositoryv2 {
     private final VendiaClient vendiaClient;
 
-    public AppointmentRepositoryv2(VendiaClient vendiaClient){
+    public AppointmentRepositoryv2(VendiaClient vendiaClient) {
         this.vendiaClient = vendiaClient;
     }
 
+    // added by Matt, please make this better!
+    public Mono<Appointment> getAppointmentById(String id) {
+        var query = new VendiaQueryBuilder()
+                .select("_id", "title", "startTime", "endTime", "location",
+                        "description", "cancelled", "participants")
+                .from("Appointment")
+                .where(new VendiaField("_id").eq(id))
+                .build();
+
+        return vendiaClient.executeQuery(query, "list_AppointmentItems", AppointmentResponse.class)
+                .map(appts -> appts.getAppointments())
+                .map(listOfAppts -> listOfAppts.get(0))
+                .doOnError(IndexOutOfBoundsException.class, (err) -> {
+                    throw new IllegalArgumentException("Invalid appointment ID: " + id);
+                });
+    }
+
     /**
      * Gets all appointments in Vendia
      *
      * @return {@link AppointmentResponse}
      */
     public Mono<AppointmentResponse> findAllAppointments() {
         var query = """
-                  query {
-                  list_AppointmentItems {
-                    _AppointmentItems {
-                      _id
-                      cancelled
-                      description
-                      endTime
-                      location
-                      participants
-                      restrictions
-                      startTime
-                      title
-                      totalSlots
-                    }
-                  }
+                	query {
+                	list_AppointmentItems {
+                		_AppointmentItems {
+                			_id
+                			cancelled
+                			description
+                			endTime
+                			location
+                			participants
+                			restrictions
+                			startTime
+                			title
+                			totalSlots
+                		}
+                	}
```

Comment:
For queries and simplicity, we should put them all in one line, i found that for integration tests this is very beneficial